tests/tests/runner.rs

```diff
@@ -33,81 +35,119 @@ use graph_core::{
-    let subgraph_name = SubgraphName::new("test1")
+async fn data_source_revert() -> anyhow::Result<()> {
+    let subgraph_name = SubgraphName::new("data-source-revert")
         .expect("Subgraph name must contain only a-z, A-Z, 0-9, '-' and '_'");
 
-    let deployment = DeploymentLocator {
-        id: DeploymentId::new(1),
-        // ethereum-blocks
-        hash: DeploymentHash::new("QmUuqWUHTNaHkq6d333n7gCrZbwzLN2yophaHeeK9x6BjJ")
-            .expect("unable to parse hash"),
+    let ipfs = IpfsClient::localhost();
+    ipfs.test()
+        .await
+        .expect("Could not connect to IPFS, make sure it's running at port 5001");
+
+    // This setup can be run once for all tests.
+    run_cmd(Command::new("yarn").current_dir("./integration-tests"));
+
+    let test_dir = format!("./integration-tests/{}", subgraph_name);
+    run_cmd(Command::new("yarn").arg("codegen").current_dir(&test_dir));
+
+    let deploy_output = run_cmd(
+        Command::new("yarn")
+            .arg("deploy:test")
+            .env("IPFS_URI", "http://127.0.0.1:5001")
+            .env("GRAPH_NODE_ADMIN_URI", "http://localhost:4242")
+            .current_dir(test_dir),
+    );
+
+    // Hack to extract deployment id from `graph deploy` output.
+    const ID_PREFIX: &str = "Build completed: ";
+    let hash = {
+        let mut line = deploy_output
+            .lines()
+            .find(|line| line.contains(ID_PREFIX))
+            .expect("found no matching line");
+        if !line.starts_with(ID_PREFIX) {
+            line = &line[5..line.len() - 5]; // workaround for colored output
+        }
+        DeploymentHash::new(line.trim_start_matches(ID_PREFIX)).unwrap()
+    };
+
+    let chain: Vec<BlockWithTriggers<Chain>> = {
+        let start_block_hash = H256::from_low_u64_be(0);
+        let start_block = BlockWithTriggers {
+            block: graph_chain_ethereum::chain::BlockFinality::Final(Arc::new(
+                LightEthereumBlock {
+                    hash: Some(start_block_hash),
+                    number: Some(U64::from(0)),
+                    ..Default::default()
+                },
+            )),
+            trigger_data: vec![],
+        };
+        let block_2 = BlockWithTriggers::<Chain> {
+            block: graph_chain_ethereum::chain::BlockFinality::Final(Arc::new(
+                LightEthereumBlock {
+                    hash: Some(H256::from_low_u64_be(1)),
+                    number: Some(U64::from(1)),
+                    parent_hash: start_block_hash,
+                    ..Default::default()
+                },
+            )),
+            trigger_data: vec![EthereumTrigger::Block(
+                test_ptr(1),
+                EthereumBlockTriggerType::Every,
+            )],
+        };
+        let block_2_reorged_hash = H256::from_low_u64_be(12);
+        let block_2_reorged = BlockWithTriggers::<Chain> {
+            block: graph_chain_ethereum::chain::BlockFinality::Final(Arc::new(
```

Comment:
Could also be useful adding a new_final_block fn to use as a shorthand for this stuff, especially if we mostly care about 3 fields most of the time

Reply:
There are nice helpers we could have for building chains, but I want to see some more tests to optimize for the patterns we see.